### ISSUE TITLE
chroma-hnswlib C++11 build error

### ISSUE THREAD

**Issue by devinbost** [langflow-ai/langflow#4541]
### Bug Description

When I attempt to build on the dockerfile in master, I'm getting this chroma-hnswlib compilation issue, even after setting the flags: `HNSWLIB_NO_NATIVE=1 CXXFLAGS="-std=c++11"`. This is a new issue as it wasn't occurring when the image was built from 1.0.18.

Stack trace below:

```
178.3   Building wheel for chroma-hnswlib (pyproject.toml): started
178.8   Building wheel for chroma-hnswlib (pyproject.toml): finished with status 'error'
178.8   error: subprocess-exited-with-error
178.8   
178.8   × Building wheel for chroma-hnswlib (pyproject.toml) did not run successfully.
178.8   │ exit code: 1
178.8   ╰─> [63 lines of output]
178.8       running bdist_wheel
178.8       running build
178.8       running build_ext
178.8       creating tmp
178.8       g++ -std=c++11 -fPIC -I/usr/local/include/python3.12 -c /tmp/tmpsoim81gv.cpp -o tmp/tmpsoim81gv.o -std=c++14
178.8       g++ -std=c++11 -fPIC -I/usr/local/include/python3.12 -c /tmp/tmpug2a7sg5.cpp -o tmp/tmpug2a7sg5.o -std=c++11
178.8       Traceback (most recent call last):
178.8         File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
178.8           main()
178.8         File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
178.8           json_out['return_val'] = hook(**hook_input['kwargs'])
178.8                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
178.8         File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 251, in build_wheel
178.8           return _build_backend().build_wheel(wheel_directory, config_settings,
178.8                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 438, in build_wheel
178.8           return _build(['bdist_wheel', '--dist-info-dir', str(metadata_directory)])
178.8                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 426, in _build
178.8           return self._build_with_temp_dir(
178.8                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 407, in _build_with_temp_dir
178.8           self.run_setup()
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 320, in run_setup
178.8           exec(code, locals())
178.8         File "<string>", line 115, in <module>
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/__init__.py", line 117, in setup
178.8           return distutils.core.setup(**attrs)
178.8                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/_distutils/core.py", line 183, in setup
178.8           return run_commands(dist)
178.8                  ^^^^^^^^^^^^^^^^^^
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/_distutils/core.py", line 199, in run_commands
178.8           dist.run_commands()
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 954, in run_commands
178.8           self.run_command(cmd)
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/dist.py", line 994, in run_command
178.8           super().run_command(command)
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 973, in run_command
178.8           cmd_obj.run()
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/command/bdist_wheel.py", line 383, in run
178.8           self.run_command("build")
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
178.8           self.distribution.run_command(command)
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/dist.py", line 994, in run_command
178.8           super().run_command(command)
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 973, in run_command
178.8           cmd_obj.run()
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/_distutils/command/build.py", line 135, in run
178.8           self.run_command(cmd_name)
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
178.8           self.distribution.run_command(command)
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/dist.py", line 994, in run_command
178.8           super().run_command(command)
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 973, in run_command
178.8           cmd_obj.run()
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/command/build_ext.py", line 99, in run
178.8           _build_ext.run(self)
178.8         File "/tmp/pip-build-env-z3voj59n/overlay/lib/python3.12/site-packages/setuptools/_distutils/command/build_ext.py", line 359, in run
178.8           self.build_extensions()
178.8         File "<string>", line 102, in build_extensions
178.8         File "<string>", line 69, in cpp_flag
178.8       RuntimeError: Unsupported compiler -- at least C++11 support is needed!
178.8       [end of output]
178.8   
178.8   note: This error originates from a subprocess, and is likely not a problem with pip.
178.8   ERROR: Failed building wheel for chroma-hnswlib
178.8 Successfully built pysher google-search-results spider-client wikipedia peewee pyperclip pypika
178.8 Failed to build chroma-hnswlib
178.8 ERROR: Could not build wheels for chroma-hnswlib, which is required to install pyproject.toml-based projects
178.8 
178.8 [notice] A new release of pip is available: 24.0 -> 24.3.1
178.8 [notice] To update, run: pip install --upgrade pip
------
failed to solve: process "/bin/sh -c HNSWLIB_NO_NATIVE=1 CXXFLAGS=\"-std=c++11\" pip install -U langflow cassio mysql-connector-python sentence-transformers langchain-huggingface langchain-astradb" did not complete successfully: exit code: 1
```

I'm on arm64 (M1 Mac).

### Reproduction

Build an image from build_and_push.Dockerfile 
Then, reference that image in the following dockerfile:

```
FROM (first image)

WORKDIR /app

RUN pip install -U langflow cassio mysql-connector-python sentence-transformers langchain-huggingface langchain-astradb

ENV LANGFLOW_LOAD_FLOWS_PATH=/app/flows

EXPOSE 7860

CMD ["langflow", "run", "--env-file", "/app/.env", "--backend-only", "--host", "0.0.0.0", "--port", "7860"]
```

### Expected behavior

Image should build just fine. 

### Who can help?

@NadirJ 

### Operating System

Mac M1 (Apple Silicon)

### Langflow Version

bleeding edge (after 1.0.19.post2)

### Python Version

3.10

**Comment by italojohnny** [comment]
Hello Devin,

From what I understand of the error you reported, the chroma-hnswlib dependency is distributed as source code and needs to be compiled in your environment. For the installation to work, you’ll need a C++ compiler that supports at least the C++11 standard. Try installing or updating a compiler like g++ or clang in your environment to allow pip to compile it properly.

Let me know if you need further assistance!

**Comment by devinbost** [comment]
Hi @italojohnny , thanks for the comments!
I actually have all of those dependencies in my host environment, but they might have been missing in the docker image. 
That said, when I removed "langflow" from the pip install step in the final build layer, the error went away, so I'm going to close this for now.

**Comment by scofano** [comment]
Is this related? I just started a brand new venv and tried to install it and got this:

Building wheels for collected packages: chroma-hnswlib
  Building wheel for chroma-hnswlib (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for chroma-hnswlib (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [9 lines of output]
      running bdist_wheel
      running build
      running build_ext
      building 'hnswlib' extension
      creating build\temp.win-amd64-cpython-312\Release\python_bindings
      "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.42.34433\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\Admin\AppData\Local\Temp\pip-build-env-x8b4srhw\overlay\Lib\site-packages\pybind11\include -IC:\Users\Admin\AppData\Local\Temp\pip-build-env-x8b4srhw\overlay\Lib\site-packages\numpy\_core\include -I./hnswlib/ -IC:\Users\Admin\AppData\Local\Programs\Python\Python312\include -IC:\Users\Admin\AppData\Local\Programs\Python\Python312\Include "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.42.34433\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.42.34433\ATLMFC\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\VS\include" /EHsc /Tp./python_bindings/bindings.cpp /Fobuild\temp.win-amd64-cpython-312\Release\./python_bindings/bindings.obj /EHsc /openmp /O2 /DVERSION_INFO=\\\"0.7.6\\\"
      bindings.cpp
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.42.34433\include\cfloat(10): fatal error C1083: NÆo ‚ poss¡vel abrir arquivo incluir: 'float.h': No such file or directory
      error: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.42.34433\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for chroma-hnswlib
Failed to build chroma-hnswlib
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (chroma-hnswlib)

**Comment by JANHMS** [comment]
same

**Comment by scofano** [comment]
How I fixed that: I installed Python 3.10, started a new venv and it worked. 
Apparently (my case) chroma could not be compiled with python 3.12
